src/libutil/source-path.hh

```diff
@@ -9,6 +9,8 @@
 #include "canon-path.hh"
 #include "source-accessor.hh"
 
+#include <boost/functional/hash.hpp> // for boost::hash_combine
+
```

Comment:
This wasn't actually used; removing in #11113 

Question though: the version in I got from boost was less capable, accepting only two arguments. Did you copy `hash_combine` from a newer version of boost than our dependency?